src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SinkIslandCheck.java

```diff
@@ -224,14 +228,17 @@ private boolean edgeCharacteristicsToIgnore(final Edge edge)
         // loop and assume that whether the check was a flag or not was handled by the other process
         return this.isFlagged(edge.getIdentifier())
                 // We don't want to handle certain types of parking amenities
-                || this.endNodeHasAmenityTypeToExclude(edge)
+                || this.endOrStartNodeHasAmenityTypeToExclude(edge)
                 // Ignore edges that have been way sectioned at the border, as has high probability
                 // of creating a false positive due to the sectioning of the way
                 || SyntheticBoundaryNodeTag.isBoundaryNode(edge.end())
                 || SyntheticBoundaryNodeTag.isBoundaryNode(edge.start())
                 // Ignore edges that are of type service and is connected to pedestrian navigable
                 // ways
-                || this.isServiceRoad(edge) && this.isConnectedToPedestrianNavigableHighway(edge);
+                || this.isServiceRoad(edge) && this.isConnectedToPedestrianNavigableHighway(edge)
+                // Ignore service roads that ends in a building or are within any airport polygon
```

Comment:
Why only service roads?

Reply:
The requirement was to remove service roads within airports. Also based on @Keahi's analysis, service roads that end up in Buildings are valid edges.